core/comment.js

```diff
@@ -104,41 +105,58 @@ Blockly.Comment.prototype.drawIcon_ = function(group) {
  * @private
  */
 Blockly.Comment.prototype.createEditor_ = function() {
-  /* Create the editor.  Here's the markup that will be generated:
+  /* Create the editor.  Here's the markup that will be generated in
+   * editable mode:
     <foreignObject x="8" y="8" width="164" height="164">
       <body xmlns="http://www.w3.org/1999/xhtml" class="blocklyMinimalBody">
         <textarea xmlns="http://www.w3.org/1999/xhtml"
             class="blocklyCommentTextarea"
             style="height: 164px; width: 164px;"></textarea>
       </body>
     </foreignObject>
-  */
+   * For non-editable mode see Warning.textToDom_.
+   */
+
   this.foreignObject_ = Blockly.utils.dom.createSvgElement('foreignObject',
       {'x': Blockly.Bubble.BORDER_WIDTH, 'y': Blockly.Bubble.BORDER_WIDTH},
       null);
+
   var body = document.createElementNS(Blockly.utils.dom.HTML_NS, 'body');
   body.setAttribute('xmlns', Blockly.utils.dom.HTML_NS);
   body.className = 'blocklyMinimalBody';
-  var textarea = document.createElementNS(Blockly.utils.dom.HTML_NS, 'textarea');
+
+  this.textarea_ = document.createElementNS(
+      Blockly.utils.dom.HTML_NS, 'textarea');
+  var textarea = this.textarea_;
   textarea.className = 'blocklyCommentTextarea';
   textarea.setAttribute('dir', this.block_.RTL ? 'RTL' : 'LTR');
+  textarea.value = this.model_.text;
+  this.resizeTextarea_();
+
   body.appendChild(textarea);
-  this.textarea_ = textarea;
   this.foreignObject_.appendChild(body);
-  Blockly.bindEventWithChecks_(textarea, 'mouseup', this, this.textareaFocus_,
+
+  // Ideally this would be hooked to the focus event for the comment.
+  // However doing so in Firefox swallows the cursor for unknown reasons.
+  // So this is hooked to mouseup instead.  No big deal.
+  Blockly.bindEventWithChecks_(textarea, 'mouseup', this, this.startEdit_,
       true, true);
   // Don't zoom with mousewheel.
   Blockly.bindEventWithChecks_(textarea, 'wheel', this, function(e) {
     e.stopPropagation();
   });
   Blockly.bindEventWithChecks_(textarea, 'change', this, function(_e) {
-    if (this.text_ != textarea.value) {
+    if (this.cachedText_ != this.model_.text) {
       Blockly.Events.fire(new Blockly.Events.BlockChange(
-          this.block_, 'comment', null, this.text_, textarea.value));
-      this.text_ = textarea.value;
+          this.block_, 'comment', null, this.cachedText_, this.model_.text));
     }
   });
+  Blockly.bindEventWithChecks_(textarea, 'input', this, function(_e) {
+    this.model_.text = textarea.value;
```

Comment:
Can we just use an ``input`` event and fire the Blockly event here?

Reply:
We could, but it would give slightly different behavior.

'input' occures on every keypress. 'change' occures when the textarea is deselected.

That's a design decision so it's up to you. (┏－‿ ◕)┏

Reply:
I see, so previously we weren't changing the ``this.text_`` property on every key press?

Reply:
That is correct.